### PR TITLE
fix(logs): add missing dyno/dynotype to datadog records

### PR DESCRIPTION
**Pull request by sileht**
This attributes are missing in datadog logs.

Change-Id: Iee17592bdc504d9523ee2e6c24b7ca77f3dd6fff

### CODE REVIEW

File: mergify_engine/logs.py [hunk 1]
```diff
@@ -72,6 +72,11 @@ def add_fields(self, log_record, record, message_dict):
         if worker_id is not None:
             log_record.update({"worker_id": worker_id})
 
+        dyno = os.getenv("DYNO", None)
```
Comment (jd) [inline mergify_engine/logs.py:75]:
```suggestion
        dyno = os.getenv("DYNO")
```

File: mergify_engine/logs.py [hunk 2]
```diff
@@ -72,6 +72,11 @@ def add_fields(self, log_record, record, message_dict):
...
+        if dyno is not None:
+            log_record.update({"dyno": dyno})
+            log_record.update({"dynotype": dyno.rsplit(".", 1)[-1]})
```
Comment (jd) [inline mergify_engine/logs.py:78]:
```suggestion
            log_record.update({"dynotype": dyno.rsplit(".", 1)[0]})
```

**Review by jd — DISMISSED** [review]

**Comment by sileht** [comment]
@mergifyio rebase

**Comment by sileht** [comment]
@Mergifyio rebase

File: mergify_engine/logs.py [hunk 3]
```diff
@@ -72,6 +72,11 @@ def add_fields(self, log_record, record, message_dict):
         if worker_id is not None:
             log_record.update({"worker_id": worker_id})
 
+        dyno = os.getenv("DYNO")
+        if dyno is not None:
+            log_record.update({"dyno": dyno})
+            log_record.update({"dynotype": dyno.rsplit(".", 1)[0]})
```
Comment (jd) [inline mergify_engine/logs.py:78]: might be worth factoring this out in a function and testing it 😉

**Review by jd — APPROVED** [review]

**Pull request merged**